### ISSUE TITLE
Gradualizer can't tell that `m` and `f` don't exist in `erlang:apply(m, f, [some, args])`

### ISSUE THREAD

**Issue by erszcz** [josefs/Gradualizer#561]
Reported by @aleklisi on erlanger.slack.com. Here's how to reproduce.

Create a new library:

```sh
r3 new lib name=mylib
```

Within it create the following module:

```sh
$ cd mylib
$ cat src/z.erl
-module(z).

f() ->
    erlang:apply(m, f, [1,2,3]).
```

Add Gradualizer as a Rebar3 plugin and run:

```sh
rebar3 gradualizer
```

Compare with:

```sh
rebar3 xref
```

The latter returns info that `m` and `f` don't exist, but Gradualizer doesn't.

**Comment by zuiderkwast** [comment]
I don't think we have any magic handling of apply and I'm not sure a type checker should do that. It's rather something for other tools, like xref. Type wise, apply is just a function that takes some atoms and a list. The type `module()` is just an alias of `atom()`.

**Comment by erszcz** [comment]
@zuiderkwast In general I agree, but I wanted to create a ticket to have a discussion around it, since this was raised by a colleague and I think it's worth considering. Given `xref` detects this, I don't think Gradualizer has to duplicate the functionality, but better have it documented that it's a deliberate choice and not just an omission.
